app/scripts/metamask-controller.js

```diff
@@ -1507,10 +1506,39 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
-      trackTransactionEvents: handleMMITransactionUpdate.bind(
-        null,
-        transactionMetricsRequest,
-      ),
+      hooks: {
+        getTransactions: this.txController.getTransactions.bind(
+          this.txController,
+        ),
+        updateTransactionHash: this.txController.setTxHash.bind(
+          this.txController,
+        ),
+        trackTransactionEvents: handleMMITransactionUpdate.bind(
+          null,
+          transactionMetricsRequest,
+        ),
+        txStateManager: {
+          getTransactions: this.txController.getTransactions.bind(
+            this.txController,
+          ),
+          setTxStatusSigned:
+            this.txController.txStateManager.setTxStatusSigned.bind(
+              this.txController.txStateManager,
+            ),
+          setTxStatusSubmitted:
+            this.txController.txStateManager.setTxStatusSubmitted.bind(
```

Comment:
Also rather than adding all `setTxStatusX` methods, we can add a generic `updateTransactionStatus` method, so it will satisfy all three methods here.

Reply:
I've included a generic status update method in the core repo. Just to clarify are you saying to add a generic also in the extension?